backend/de.metas.dunning/src/main/java/de/metas/dunning/api/impl/DunningBL.java

```diff
@@ -247,6 +253,59 @@ public void processCandidates(final IDunningContext context, final IDunningCandi
 		}
 
 		dunningProducer.finish();
+
+		sendMassNotifications(dunningProducer.getCreatedDunningDocIds());
+	}
+
+	private void sendMassNotifications(@NonNull final Collection<DunningDocId> createdDunningDocIds)
+	{
+		dunningDAO.getByIdsInTrx(createdDunningDocIds)
+				.stream()
+				.collect(Collectors.groupingBy(doc -> docTypeDAO.loadById(DocTypeId.ofRepoId(doc.getC_DocType_ID())).getMassGenerateNotification()))
+				.forEach(this::sendMassNotifications);
+	}
+
+	private void sendMassNotifications(@NonNull final BoilerPlateWithLineId boilerPlateWithLineId, @NonNull final List<I_C_DunningDoc> dunningDocs)
+	{
+		dunningDocs.stream()
+				.collect(Collectors.groupingBy(doc -> OrgId.ofRepoId(doc.getAD_Org_ID())))
+				.forEach((orgId, docs) -> this.sendMassNotifications(boilerPlateWithLineId, orgId, docs));
+	}
+
+	private void sendMassNotifications(final @NonNull BoilerPlateWithLineId boilerPlateWithLineId,
+			@NonNull final OrgId orgId,
+			@NonNull final List<I_C_DunningDoc> dunningDocs)
+	{
+		final Language userLanguage = Env.getLanguage();
+		final BoilerPlate headerBoilerPlate = boilerPlateRepository.getByBoilerPlateId(boilerPlateWithLineId.getHeaderId(), userLanguage);
+		final String lines = getMassGenerateLinesPlainString(boilerPlateWithLineId, dunningDocs);
+
+		final Evaluatee headerEvaluatee = Evaluatees.ofSingleton(MASS_GENERATE_LINES_EVALUATEE_PARAM, lines)
+				.andComposeWith(getPO(orgDAO.getById(orgId)));
+
+		notificationBL.send(UserNotificationRequest.builder()
+				.notificationGroupName(MASS_DUNNING_NOTIFICATION_GROUP_NAME)
+				.recipient(Recipient.allOrgUsersForGroupAndOrgId(MASS_DUNNING_NOTIFICATION_GROUP_NAME, orgId))
+				.contentPlain(headerBoilerPlate.evaluateTextSnippet(headerEvaluatee))
+				.subjectPlain(headerBoilerPlate.evaluateSubject(headerEvaluatee))
+				.eMailCustomType(EMailCustomType.MassDunning)
+				.build());
+	}
+
+	@NonNull
+	private String getMassGenerateLinesPlainString(final @NonNull BoilerPlateWithLineId boilerPlateWithLineId, final @NonNull List<I_C_DunningDoc> dunningDocs)
+	{
+		final Language userLanguage = Env.getLanguage();
```

Comment:
why "maybe" logged in language and not Language.getBaseAD_Language()?
anyways, I would introduce String adLanguage as a param and I would defer this decision as much as possible.

Reply:
moved the language as a param